Entities/AtomGroup.h

```diff
@@ -1,662 +1,387 @@
+		/// </summary>
+		/// <param name="moidToIgnore">The MOID to add to the ignore list.</param>
+		void AddMOIDToIgnore(MOID moidToIgnore) { m_IgnoreMOIDs.push_back(moidToIgnore); }
+
+		/// <summary>
+		/// Checks whether this AtomGroup is set to ignore collisions with a MO of a specific MOID.
+		/// </summary>
+		/// <param name="whichMOID">The MOID to check if it is ignored.</param>
+		/// <returns>Whether or not this MOID is being ignored.</returns>
+		bool IsIgnoringMOID(MOID whichMOID) { return (*(m_Atoms.begin()))->IsIgnoringMOID(whichMOID); }
+
+		/// <summary>
+		/// Clears the list of MOIDs that this AtomGroup is set to ignore collisions with during its next Travel sequence.
+		/// This should be done each frame so that fresh MOIDs can be re-added. (MOIDs are only valid during a frame).
+		/// </summary>
+		void ClearMOIDIgnoreList() { m_IgnoreMOIDs.clear(); }
+
+		/// <summary>
+		/// Gets whether any of the Atoms in this AtomGroup are on top of terrain pixels.
+		/// </summary>
+		/// <returns>Whether any Atom of this AtomGroup is on top of a terrain pixel.</returns>
+		bool InTerrain() const;
+
+		/// <summary>
+		/// Gets the ratio of how many Atoms of this AtomGroup that are on top of intact terrain pixels.
+		/// </summary>
+		/// <returns>The ratio of Atoms on top of terrain pixels, form 0 to 1.0.</returns>
+		float RatioInTerrain() const;
+
+		/// <summary>
+		/// Checks whether any of the Atoms in this AtomGroup are on top of terrain pixels, and if so, attempt to move the OwnerMO out so none of the Atoms are inside any terrain pixels anymore.
+		/// </summary>
+		/// <param name="position">Current position of the owner MO.</param>
+		/// <param name="strongerThan">Only attempt to move out of materials stronger than this specific ID.</param>
+		/// <returns>Whether any intersection was successfully resolved. Will return true even if there wasn't any intersections to begin with.</returns>
+		bool ResolveTerrainIntersection(Vector &position, unsigned char strongerThan = 0) const;
+
+		/// <summary>
+		/// Checks whether any of the Atoms in this AtomGroup are on top of MOSprites, and if so, attempt to move the OwnerMO out so none of the Atoms are inside the other MOSprite's silhouette anymore.
+		/// </summary>
+		/// <param name="position">Current position of the owner MO.</param>
+		/// <param name="rotation">Current rotation of the owner MO.</param>
+		/// <returns>Whether all intersections were successfully resolved.</returns>
+		bool ResolveMOSIntersection(Vector &position);
+#pragma endregion
+
+#pragma region Debug
+		/// <summary>
+		/// Draws this AtomGroup's current graphical debug representation to a BITMAP of choice.
+		/// </summary>
+		/// <param name="targetBitmap">A pointer to a BITMAP to draw on.</param>
+		/// <param name="targetPos">The absolute position of the target bitmap's upper left corner in the Scene.</param>
```

Comment:
If this is absolute scene position, why would we ever use the default value here? Things are pretty much never drawn at exactly (0, 0). Are there any uses of this with the default value? If not, kill off the default for this and leave it mandatory.

Alternatively, if this is actually offset, change the variable name and comments accordingly.